src/groups/semidirect_product_group.jl

```diff
@@ -272,6 +272,130 @@ function _compose!(
     return k
 end
 
+_doc_LSDP_diff_left_compose = """
+    diff_left_compose(
+        SDPG::LieGroup{𝔽,LeftSemidirectProductGroupOperation,<:ProductManifold}, g, h, X
+    ) where {𝔽}
+    diff_left_compose!(
+        SDPG::LieGroup{𝔽,LeftSemidirectProductGroupOperation,<:ProductManifold}, Y, g, h, X
+    ) where {𝔽}
+
+Compute the differential of the left group operation ``λ_g``, that is ``D_{λ_g}(h)[X]``.
+For this case it is given by
+
+```math
+    D_{λ_g}(h)[X] = $(_tex(:bigl))( D_{λ_{g_1}}(h_1)[X_1], D_{λ_{g_2}}(σ_{g_1}(h_2))$(_tex(:bigl))[ D_{σ_{g_1}}(h_2)[X_2]$(_tex(:bigr))]$(_tex(:bigr)))
+```
+"""
+
+"$(_doc_LSDP_diff_left_compose)"
+diff_left_compose(
+    SDPG::LieGroup{𝔽,<:LeftSemidirectProductGroupOperation,<:ProductManifold}, g, h, X
+) where {𝔽}
+
+"$(_doc_LSDP_diff_left_compose)"
+function diff_left_compose!(
+    SDPG::LieGroup{𝔽,<:LeftSemidirectProductGroupOperation,<:ProductManifold}, Y, g, h, X
+) where {𝔽}
+    PM = SDPG.manifold
+    G, H = map(LieGroup, PM.manifolds, SDPG.op.operations)
+    A = GroupAction(SDPG.op.action_type, G, H)
```

Comment:
I think we should split this out to a separate function to reduce code duplication. It's already in 4 other places and I suspect they don't all get the order of G and H correct (it changes between left and right semidirect product).

Reply:
do you have a good idea for a name?

Reply:
concerning the order - for me this takes a lot of thinking and scribbling a few formulae somewhere to implement these, so I hope they are in the right order.

Reply:
Maybe `semidirect_group_action(SPDG)` ?

The order of G and H is inconsistent between different methods so I'm a bit suspicious. Anyway, refactoring it to a new function will fix any issues that might be there.

Reply:
Will do that, maybe during the day if I find a. bit of time. I am tending a bit towards a name more like `semidifrect_group_elements` or so since it returns more that just the action.